src/connectRoutes.js

```diff
@@ -126,6 +130,14 @@ export default (
     initialDispatch: shouldPerformInitialDispatch = true
   }: Options = options
 
+  const selectLocationState = typeof locationKey === 'function'
+    ? locationKey
+    : state => state[locationKey]
+
+  const selectTitleState = typeof titleKey === 'function'
+    ? titleKey
+    : state => state[titleKey]
+
```

Comment:
that too, but u often see the pattern of being able to provide a string or a selector. it's not uncommon even in 2017.